src/apps/ocioconvert/main.cpp

```diff
@@ -625,6 +625,11 @@ int main(int argc, const char **argv)
             exit(1);
         }
 
+        if (outputcolorspace)
+        {
+            spec.attribute("oiio:ColorSpace", outputcolorspace);
+        }
+
```

Comment:
You could improve this by adding an "else if (useDisplayView)" block that basically does this:
outputcolorspace = getDisplayViewColorSpaceName(display, view);

Reply:
Sorry I'm not familiar with the detailed API. The issue and PR here are just to generate a "proper enough" JPG||PNG with colorspace, so that with most of system default viewers could display the image correctly.

The display and view are 2 strings, so I'd prefer to set 2 extra metadata fields `ocio:display` and `ocio:view` to hold the vlaues.

But for `outputcolorspace`, this is single value to allow OIIO to generate actual file, what's the best value ? Display or view ? I prefer the value of display, it's more generic than view.

WDYT ?

Reply:
I like your goal of having "most of system default viewers could display the image correctly", but unfortunately I don't follow how this would work.  The name being written here is the name from the user's OCIO config file.  These names are not standardized.  So how would the "system default viewers" act on this information?  I guess you are referring to operating system included viewers such as Preview on MacOS?  As far as I know, none of these are using OCIO, so you would need to tag the image with a string that would be meaningful to these other apps.  Is there a standard for that that you had in mind?  Do you have a prototype of this working in some app?

Regarding your suggestion of adding 2 new metadata fields for ocio:display and ocio:view, could you elaborate on how you expect these would be used by downstream software and which file formats would be able to make use of this information?

As I wrote, you should use getDisplayViewColorSpaceName to set the outputcolorspace for the oiio:ColorSpace field.  As I wrote in the issue, the display is more relevant than the view, but I would not use the raw display string since this may not actually be a color space.  Furthermore, in many configs it is not even used as a "display".  For example, in the popular 1.x ACES configs it is "ACES".

Thanks!

Reply:
`
I like your goal of having "most of system default viewers could display the image correctly", but unfortunately I don't follow how this would work. The name being written here is the name from the user's OCIO config file. These names are not standardized. So how would the "system default viewers" act on this information? I guess you are referring to operating system included viewers such as Preview on MacOS? As far as I know, none of these are using OCIO, so you would need to tag the image with a string that would be meaningful to these other apps. Is there a standard for that that you had in mind? Do you have a prototype of this working in some app?
`
I think you didn't try the issue #1420, please have a try at different platforms with and without modification, then you would see the difference. As the issue reported, specified sRGB as output colorspace, it shouldn't write Linear as metadata.

`
Regarding your suggestion of adding 2 new metadata fields for ocio:display and ocio:view, could you elaborate on how you expect these would be used by downstream software and which file formats would be able to make use of this information?
`
This is not going to be used by any software for now, just as metadata for the generated image, just as EXIF information when you take a photo. Assuming user generates several different PNGs from single EXR with difference display+view, metadata would be safer and easier for debugging purpose. But I think this should be done from another issue/PR as a feature.

`
As I wrote, you should use getDisplayViewColorSpaceName to set the outputcolorspace for the oiio:ColorSpace field. As I wrote in the issue, the display is more relevant than the view, but I would not use the raw display string since this may not actually be a color space. Furthermore, in many configs it is not even used as a "display". For example, in the popular 1.x ACES configs it is "ACES".
`
This is fine. I did the commit.

https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1500/commits/beccd66e3d1974193e18817c825eb7584f5057ee

Thanks.

Reply:
> Regarding your suggestion of adding 2 new metadata fields for ocio:display and ocio:view, could you elaborate on how you expect these would be used by downstream software and which file formats would be able to make use of this information?

Providing both the display and view permits downstream software to defer the choice of how "far" to invert the baked-in rendering to the user; i.e., whether to decode to an output-referred image state or a scene-referred state (render-time look / data overrides notwithstanding, I suppose).  Even if the view inversion is potentially lossy, there are absolutely cases where the need to work with scene-referred relative exposure values trumps the need for 100% faithful reproduction. 

Display / View inversion is also extremely convenient for previewing multiple media sources of differing image states under a single user-selected Display / View (which, under the right conditions, might end up bypassing any lossy inversions along the way). 

Also, as Bo says, persisting both Display and View metadata is extremely useful for debugging. In a world where everyone is working remotely, with potential technical and legal constraints on one's ability to share what they're seeing and how they're viewing, when someone says "Hey, my dailies renders aren't matching the editorial reference", or "Something changed, and now everything looks wrong", being able to quickly ascertain both the Display and View used to produce the image in question (without having to dig through render logs or earlier versions of comp scripts) is worth its weight in gold. 

Lastly, always considering only the Display will likely produce unexpected and inconsistent behavior across v1- and v2-style configs, where imagery produced with v1 configs will always decode to a scene-referred state by default, while images produced with v2-style Display View transforms will always decode to a display-referred state. Personally, for this reason alone, I'd have expected downstream software's decoding behavior to default to fully inverting both the View and the Display, as default behavior discrepancies can be extremely difficult to track down and debug.

Just my two cents : )

Reply:
Thanks for the update.  When you wrote "most system default viewers" were you referring to ImageMagick?  I'm curious if this fix also improves behavior in any other viewers you tested.

Reply:
Zach, you raise interesting points, but it seems you are assuming the downstream software will be using OCIO, which seems to be a different use-case than what zhoub had in mind.  If the intent is to write separate display and view strings with the intent of accurately inverting this downstream, perhaps the config path and context variables also need to be saved for that to be meaningful?  In any case, I have no objections to adding display, view and other descriptive metadata like that, even with the understanding that it won't make it into most formats (OpenEXR only?).

Having faced many issues in the past with image format metadata being interpreted differently by downstream apps, I find the concept of inverting a display/view to be incredibly ambitious and still feel that it would be more realistic to simply get apps to agree on how to display the current image correctly on a monitor.  :)  But to be clear, the current PR does not even do that, all it does is prevent OIIO from incorrectly tagging it as "Linear".

Reply:
I thought your comments added some interesting considerations that I hadn't thought of, so thank you for the posts Zach (definitely not a derailment!).  

I agree with you that the PR is ready to be merged but was waiting to hear if there are any other comments from people.  Also would be interested to hear more from Bo about which apps he tested his work in.

Reply:
For reference purposes, and because I mentioned it on Slack here is what the VES Transfer/Interchange Spec recommends:

![image](https://user-images.githubusercontent.com/99779/135923720-36aebe4f-66a1-4b70-bbaa-cd6a8a795511.png)